pyccel/ast/core.py

```diff
@@ -2569,6 +2569,14 @@ def body(self):
         """ CodeBlock containing all the statements in the function """
         return self._body
 
+    @body.setter
+    def body(self, body):
+        if iterable(body):
+            body = CodeBlock(body)
+        elif not isinstance(body, CodeBlock):
+            raise TypeError('body must be an iterable or a CodeBlock')
+        self._body = body
```

Comment:
```suggestion
        self._body.invalidate_node()
        self._body = body
        self._body.set_current_user_node(self)
```

Reply:
@sboof911 Sorry I just noticed that I had a line wrong here

Reply:
Fixed.